test/integration/clientHelper.ts

```diff
@@ -19,14 +23,26 @@ export async function registerUser(username: string, displayname: string, passwo
     let { data } = await axios.get(registerUrl);
     let nonce = data.nonce!;
     let mac = HmacSHA1(`${nonce}\0${username}\0${password}\0${admin ? 'admin' : 'notadmin'}`, 'REGISTRATION_SHARED_SECRET');
-    return await axios.post(registerUrl, {
-        nonce,
-        username,
-        displayname,
-        password,
-        admin,
-        mac: mac.toString()
-    })
+    for (let i = 1; i <= REGISTRATION_ATTEMPTS; ++i) {
+        try {
+            return await axios.post(registerUrl, {
```

Comment:
not worth it for this patch at all but i wonder if we could have avoided using `axios` all together by using bot-sdk's request stuff and remove an extra dependency

Reply:
Good point. Could be.